testmocks/src/main/java/org/apache/zookeeper/MockZooKeeper.java

```diff
@@ -144,10 +144,7 @@ public static MockZooKeeper newInstanceForGlobalZK(ExecutorService executor, int
 
     public static MockZooKeeper newInstance(ExecutorService executor, int readOpDelayMs) {
         try {
-            MockZooKeeper zk = createMockZooKeeperInstance(executor, readOpDelayMs);
-            ObjectInstantiator<ClientCnxn> clientCnxnObjectInstantiator = objenesis.getInstantiatorOf(ClientCnxn.class);
-            Whitebox.setInternalState(zk, "cnxn", clientCnxnObjectInstantiator.newInstance());
-            return zk;
+            return createMockZooKeeperInstance(executor, readOpDelayMs);
```

Comment:
@eolivelli this change was added by you in https://github.com/apache/pulsar/pull/8590. I don't know why we need this `WhiteBox` based mock and after removing it and running several tests, it seems all green.

Could you share some background here?

Reply:
Maybe the older versions of Mockito or other mock frameworks needed it